go/libkb/ping_gregor.go

```diff
@@ -0,0 +1,144 @@
+	return t.stagedTransport, nil
+}
+
+func (t *pingGregorTransport) IsConnected() bool {
+	t.G().Log.Debug("pingGregorTransport IsConnected")
+	return t.transport != nil && t.transport.IsConnected()
+}
+
+func (t *pingGregorTransport) Finalize() {
+	t.G().Log.Debug("pingGregorTransport Finalize")
+	t.transport = t.stagedTransport
+	t.stagedTransport = nil
+}
+
+func (t *pingGregorTransport) Close() {
+	t.G().Log.Debug("pingGregorTransport Close")
+	t.conn.Close()
+}
+
+// pingGregorHandler implements rpc.ConnectionHandler
+type pingGregorHandler struct {
+	Contextified
+	pingErrors  chan error
+	pingSuccess chan struct{}
+}
+
+var _ rpc.ConnectionHandler = (*pingGregorHandler)(nil)
+
+func newGregorHandler() *pingGregorHandler {
+	return &pingGregorHandler{}
+}
+
+func (g *pingGregorHandler) HandlerName() string {
+	return "ping gregor"
+}
+
+func (g *pingGregorHandler) OnConnect(ctx context.Context, conn *rpc.Connection, cli rpc.GenericClient, srv *rpc.Server) error {
+	g.G().Log.Debug("pingGregorHandler OnConnect")
+	ac := gregor1.IncomingClient{Cli: cli}
+	response, err := ac.Ping(ctx)
+	if err != nil {
+		g.pingErrors <- err
+	} else if response != "pong" {
+		g.pingErrors <- fmt.Errorf("Got an unexpected response from ping: %#v", response)
+	} else {
+		g.pingSuccess <- struct{}{}
+	}
+	return err
+}
+
+func (g *pingGregorHandler) OnConnectError(err error, reconnectThrottleDuration time.Duration) {
+	g.G().Log.Debug("pingGregorHandler OnConnectError", err)
+	g.pingErrors <- err
+}
+
+func (g *pingGregorHandler) OnDisconnected(ctx context.Context, status rpc.DisconnectStatus) {
+	g.G().Log.Debug("pingGregorHandler OnDisconnected", status)
+}
+
+func (g *pingGregorHandler) OnDoCommandError(err error, nextTime time.Duration) {
+	g.G().Log.Debug("pingGregorHandler DoCommandError", err)
+	g.pingErrors <- err
+}
+
+func (g *pingGregorHandler) ShouldRetry(name string, err error) bool {
+	g.G().Log.Debug("pingGregorHandler ShouldRetry", name, err)
+	g.pingErrors <- err
+	return false
+}
+
+func (g *pingGregorHandler) ShouldRetryOnConnect(err error) bool {
+	g.G().Log.Debug("pingGregorHandler ShouldRetryOnConnect", err)
+	g.pingErrors <- err
+	return false
+}
+
+func (g *pingGregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
+	g.G().Log.Debug("pingGregorHandler BroadcaseMessage", m)
+	return nil
+}
+
+func PingGregor(g *GlobalContext) error {
+	var pingError error
+	g.Log.Debug("+ libkb.PingGregor")
+	defer g.Log.Debug("- libkb.PingGregor %#v", pingError)
+
+	transport := &pingGregorTransport{
+		Contextified: NewContextified(g),
+		host:         strings.TrimPrefix(g.Env.GetGregorURI(), "fmprpc://"),
+	}
+	rpcHandler := &pingGregorHandler{
+		Contextified: NewContextified(g),
+		pingErrors:   make(chan error),
+		pingSuccess:  make(chan struct{}),
+	}
+	connection := rpc.NewConnectionWithTransport(rpcHandler, transport, keybase1.ErrorUnwrapper{}, true, keybase1.WrapError, g.Log, nil)
```

Comment:
It's possible that `NewConnectionWithTransport` was the wrong interface to use here, and that I should've used a simpler thing that wouldn't have needed all those methods above. If we want to extend this to test authentication later, though, I think we'll need a full connection and not just a one-off RPC?